### PR TITLE
Manual update

### LINKED ISSUE #1073: Document the LaTeXML Daemon at top-level of LaTeXML manual
The LaTeXML manual only talks about `latexml` and `latexmlpost` in Section/Chapter 2 "Using LaTeXML". This should really have a top-level section about the motivation and practicalities of `latexmls` and `latexmlc`. Nothing like this seems to exist at the moment.

### LINKED ISSUE #1402: latexmlpost --format=xml does not seem to convert <Math> to MathML
The documentation for the `--format=xml` option says:

> For the default, xml, the output is left in LaTeXML’s internal xml, but the math is parsed and converted to presentation MathML.

However this doesn't appear to happen in fact.

For example:

    % cat try.tex
    \documentclass{book}
    \title{A book}
    \author{By me}
    \begin{document}
    \maketitle

    Hello, here is an equation: $E=mc^2$.
    \end{document}
    % latexml --dest=try.lxml try.tex
    latexml (LaTeXML version 0.8.4)
    ...
    % latexmlpost --format=xml --destination=try.xmlout try.lxml
    latexmlpost (LaTeXML version 0.8.4)
    ...
    % grep Math try.xmlout
    <p>Hello, here is an equation: <Math mode="inline" tex="E=mc^{2}" text="E = m * c ^ 2" xml:id="p1.m1" fragid="p1.m1">
        <XMath>
        </XMath>
      </Math>.</p>
    %

Using `--format=xhtml` does work as expected, so I don't think this is pointing to some grievous breakage in my installation (most recent version from macports).

My use-case here is that I (probably) want to use my own stylesheets to convert from the LaTeXML internal format to XHTML, but with the conversion to MathML done for me (the reason for that is that the conversion with `--format=xhtml` does 95% of what I want – LaTeXML is a wonderful piece of work! – but the last 5% is dissolving into a mess of special cases, which are messy enough that I suspect it'd be easier overall if I start my post-post-processing a little earlier in the toolchain).

### LINKED ISSUE #1416: docs: build from source
https://dlmf.nist.gov/LaTeXML/get.html could usefully link to 
https://lists.informatik.uni-erlangen.de/pipermail/latexml/2018-May/002368.html
in the Prerequisites section.

### PR DESCRIPTION
**Pull request by brucemiller**
An update to the Manual ( and site).

First patches, just for smooth building and stability.

fixes #1073
fixes #1402
fixes #1416

### CODE REVIEW

**Comment by dginev** [comment]
Could you also include your recommended command for building the manual locally? That way I can reproduce over here and check the rendered variants as well.

**Comment by brucemiller** [comment]
Using `doc/makesite` will generally make the site pages, as well as the manual pdf and pages; a few options to enable/disable.  You'll want `--webdir=/someplace` to tell it where to put the web pages.

**Comment by tkw1536** [comment]
I'm unable to run makesite on a blank directory:
```
$ mkdir /Users/twiesing/tmp/empty-webdir
$ ./makesite --webdir=/Users/twiesing/tmp/empty-webdir
==================================================
Generating site pages
Couldn't read directory /Users/twiesing/tmp/empty-webdir/releases: No such file or directory at ./makesite line 113.
 at ./makesite line 113.
        main::getReleases() called at ./makesite line 55
```

**Comment by brucemiller** [comment]
more updates;

fixes #1216
fixes #1073 
fixes #1402 
fixes #1416 

And nibbles at the edges of #1213

**Comment by dginev** [comment]
Why/how does this PR fix #1216 ? I thought the only way out of that was some system-level change from the OS, or us using something different altogether.

**Comment by brucemiller** [comment]
As I understood it, #1216 was about documenting how to install 
dependencies when you want to run the development version of LaTeXML.
(But then I didn't understand where mktexlsr came in, so perhaps
I don't understand exactly which situation the linked "recipe" solves)

**Comment by dginev** [comment]
> As I understood it, #1216 was about documenting how to install 
> dependencies when you want to run the development version of LaTeXML.
> (But then I didn't understand where mktexlsr came in, so perhaps
> I don't understand exactly which situation the linked "recipe" solves)
> 

Well that issue is rather annoying for lay users to the point where it is unclear if documenting how to mod a file under /etc really "solves" the usability problem. I'd rather keep it open until we find something a bit more satisfying...

**Comment by brucemiller** [comment]
Oh! Too many issues; I thought you were referring to 1416, not 1216.

Well, the solution there is simply to use newer versions of ImageMagic 
and/or Ghostscript. The documentation change just points that out.

**Comment by brucemiller** [comment]
OK, the last commit should make it a bit more friendly for casual use.

**Comment by brucemiller** [comment]
This PR addresses some documentation issues, but isn't a massive rewrite. I think in the next go round, a separate set of howto documents, as suggested in #1213 would be a good idea (although details of how to write/build it are tbd); this would also simplify the manual.

**Comment by dginev** [comment]
> Well, the solution there is simply to use newer versions of ImageMagic 
> and/or Ghostscript. The documentation change just points that out.

I don't follow. I thought I saw the problematic ghostscript security policy on the latest Ubuntu release as well? Could you point me to the part of the PR explaining the details?

File: doc/site/index.tex [hunk 1]
```diff
@@ -541,67 +592,78 @@ \subsection{CPANM Installation}\label{get.cpanm}
...
-
-% Possibly, eventually, want to expose:
-%   https://www.mathweb.org/wiki/????
-% But, it doesn't have anything in it yet.
+\paragraph*{ImageMagick Security} is also an issue, due to the possibility of embadding
+malicious code in postscript, pdf and other image formats.
+For a certain range of versions of ImageMagic and Ghostscript, a security policy was set
+which inhibited processing of pdf files; the problem appears to be fixed recent versions.
+If you see \texttt{operation not allowed by the security policy}, see the discussion
+at \href{https://stackoverflow.com/a/53180170}{stackoverflow}.
```
Comment (brucemiller) [inline doc/site/index.tex:650]: @dginev This is where I address the issue, and suggest using the most recent versions, if you can. I also point to the stack overflow discussion where you can figure out if you need to do something special and how to do it.  It is way out of scope here to get into explaining how to modify the policy files of random OS's, however.

Comment (dginev) [reply doc/site/index.tex:650]: Ok, so the security problem in ghostscript is fixed in recent versions, but the actual usability problem remains - latexml is broken by default for image conversion in ubuntu/debian systems (at the least).

All I ask is that we keep the issue in question open until the default state of image conversion on those OSes is "working", however we manage to achieve that.

**Comment by brucemiller** [comment]
Two solutions are given: update to a not broken by default system, or 
follow the linked instructions.

**Comment by dginev** [comment]
> Two solutions are given: update to a not broken by default system

To my understanding the latest versions of debian, ubuntu and archlinux all have the security  policy in question that makes image conversion broken by default. So you cant update to a working state from multiple distros. I have not checked others, but to my perspective closing the image conversion issue with this PR is pretty much a "worksforme" resolution.

**Comment by brucemiller** [comment]
Image processing by LaTeXML is clearly documented as optional; And, as 
clearly documented, it *REQUIRES* a *WORKING* ImageMagick or 
Graphics::Magick.

If that makes LaTeXML "broken by default" on systems that are "broken by 
default", well, so be it. If you can't upgrade your "broken by default" 
system to a not "broken by default" system, you can modify
the "broken by default" security policy, as outlined in the linked 
stackoverflow.

Writing our own version of ImageMagick and postscript/pdf interpreter is 
not practical, certainly not "best practice", no other workaround within 
LaTeXML seems feasible. Solutions have been clearly documented in the 
installation guide and in a to-be-closed, but findable, Issue. So 
there's no point in keeping an issue open indefinitely waiting for 
action by some third party. That's not "worksforme".

Done.

Any constructive comments on the PR?

File: doc/site/index.tex [hunk 2]
```diff
@@ -464,37 +541,13 @@ \subsubsection{Windows using Strawberry Perl}\label{get.windows.strawberry}
 You may want to try \href{https://www.imagemagick.org/}{ImageMagick}, but
 you're on your own, there!  You may  have better luck with \texttt{Graphics::Magick}.
 
-\paragraph*{Installing \emph{only} prerequisites  under Strawberry}\label{get.windows.strawberry.prereq}\\
-Install \href{https://strawberryperl.com}{Strawberry Perl}, and
-the \TeX-system of your choice (if desired), 
-ImageMagick (if you want to try; see \ref{get.windows.imagemagick}),
-and then install the additional prerequisites as
+\paragraph*{Installi \emph{only} prerequisites  under Strawberry,}\label{get.windows.strawberry.prereq}
+except for \TeX, and ImageMagic (see \ref{get.windows.imagemagick}),
```
Comment (dginev) [inline doc/site/index.tex:545]: a bunch of `ImageMagic` typos for `ImageMagick` with a trailing `k`, probably better to grep than for me to comment on each one.

File: lib/LaTeXML/Common/Model/RelaxNG.pm [hunk 3]
```diff
@@ -551,6 +551,7 @@ sub cleanTeX {
   $string =~ s/\#/\\#/g;
   $string =~ s/<([^>]*)>/\\texttt{$1}/g;    # An apparent convention <sometext> == ttfont?
   $string =~ s/_/\\_/g;
+  $string =~ s/\*/any/g;                    # ????
```
Comment (dginev) [inline lib/LaTeXML/Common/Model/RelaxNG.pm:554]: unsure about this comment, what is a sample input that has the star?

File: lib/LaTeXML/Common/XML.pm [hunk 4]
```diff
@@ -328,3 +328,103 @@ BEGIN {
...
+
+Returns the list of values for the given attributes on C<$node>
+
+=item C<initialize_catalogs()>
+
+Initialize XML::LibXML to recognize the catalogs given in LaTeXML.catalogs.
+
+=item C<set_RDFa_prefixes($document,$map)>
+
+This needs an explanation.
```
Comment (dginev) [inline lib/LaTeXML/Common/XML.pm:416]: Maybe:
"This method scans the document's RDFa attributes, extracting the prefixes used. These prefixes are then filtered through a `$map` of known RDFa prefixes and the ones allowed are declared globally for the document via the "prefix" attribute of its root element."

**Review by dginev — CHANGES_REQUESTED** [review]
Great PR overall!

I spotted some minor edits, good to merge right away apart from those.

File: lib/LaTeXML/Common/Model/RelaxNG.pm (hunk 3 above)
Comment (brucemiller) [reply lib/LaTeXML/Common/Model/RelaxNG.pm:554]: it's for `rawhtml` which can take *any* xhtml.

File: doc/site/index.tex (hunk 2 above)
Comment (brucemiller) [reply doc/site/index.tex]: fixed.

File: lib/LaTeXML/Common/XML.pm (hunk 4 above)
Comment (brucemiller) [reply lib/LaTeXML/Common/XML.pm:416]: done; thanks!

**Pull request merged**